### DISCUSSION TITLE
Failed to open stream: Too many open files.

### DISCUSSION THREAD

**Discussion by giovanniledda** [spatie/laravel-backup#1729] [Q&A]
Running a backup on a large project filesystem, here is what I get:

```
Starting backup...
Dumping database storm...
Determining files to backup...
Zipping 24745 files and directories...
Created zip containing 24745 files and directories. Size is 2.23 GB
Copying zip to disk named s3-backups...
Copying zip failed because: Unable to open "/Users/zoba-mac/Sites/laravel/storm-laravel/storage/app/backup-temp/temp/2023-10-30-17-56-11.zip" using mode "r": fopen(/Users/zoba-mac/Sites/laravel/storm-laravel/storage/app/backup-temp/temp/2023-10-30-17-56-11.zip): Failed to open stream: Too many open files.
Backup failed because include(/Users/zoba-mac/Sites/laravel/storm-laravel/vendor/spatie/laravel-backup/src/Events/BackupHasFailed.php): Failed to open stream: Too many open files.
```

**Comment by niels-numbers** [comment]
Sorry, cannot help you with your current problem. But I have a similar project with same about of files and size. I run into the following problems:

1) When the process backup:run is triggered, my server is extremly slow, almost unusable. Even had requests from users that they got timeout issue when trying to download something from my server. This makes me wonder, if I even can do the backup
2) I either get "ZipArchive::close(): Can't open file: No such file or directory" or no error at all, just a silent fail without logs https://stackoverflow.com/questions/77401220/ziparchiveclose-cant-open-file-okay-but-which-file

I am starting to wonder, if this package was even created for a project with so many files & storage.

**Comment by clicktodev** [comment]
I'd appreciate it if @freekmurze could provide some insight into why this was turned into a discussion since it seems to be a valid issue .

**Comment by danharper83** [comment]
I'm getting the same issue.
